seqr/utils/file_utils.py

```diff
@@ -81,17 +81,14 @@ def mv_file_to_gs(local_path, gs_path, user=None):
     _run_gsutil_with_wait(command, gs_path, user)
 
 
-def get_gs_file_list(gs_path, user=None):
+def get_gs_file_list(gs_path, user=None, recursive=True):
     gs_path = gs_path.rstrip('/')
     command = 'ls'
 
-    # If a bucket is empty gsutil throws an error when running ls with ** instead of returning an empty list
-    subfolders = _run_gsutil_with_wait(command, gs_path, user, get_stdout=True)
-    if not subfolders:
-        return []
```

Comment:
This is unnecessary after the error handling of `_run_gsutil_with_wait` was improved.

Reply:
The `_run_gsutil_with_wait` function raises an exception for the `process.wait() != 0`. Your comment reminds me to add exception capturing in the API. The concern will be that there may still be warning messages when the `process.wait() == 0`. Can we add a `-q` option to the `gsutil` command to silence any warnings?

Reply:
As the comment that you deleted here says `If a bucket is empty gsutil throws an error when running ls with **`. This has nothing to do with our error handling, and it has nothing to do with silencing warnings. Unless the gsutil behavior has changed and no longer throws an error in that case, we need to keep this code as is

Reply:
The behavior doesn't change. But we should handle the case as an error because an empty bucket provides no data for analysis. So I added exception capturing in the API code to handle the error.

Reply:
I really disagree with making this change. I do not understand why its necessary, it was added for a very specific reason and as far as I can tell there is no reason to believe its no longer needed. Please explain why you think we need this change because from my perspective this is a risky change that has no relationship to the functionality change this PR is supposed to be working on

Reply:
We have to handle the exception of `process.wait() != 0` in the `validate_anvil_vcf` and `get_anvil_vcf_list` APIs. We should have added the exception handling when we added raising an exception for `process.wait() != 0`. Once we handle the exceptions in the APIs, we don't have to run an extra `gsutil` command to avoid exceptions.

Reply:
> we don't have to run an extra gsutil command to avoid exceptions.

This is out of scope of this PR. This is an optimization that is risky that is having a cascading effect on a lot of other change and is making it really difficult to figure out what is changing because we actually need the change to support sharded VCFs, and what is changing because of this optimization. If you feel really strongly that we need this optimization, that needs to go in a separate PR. However, I do not think that is a good use of your time. The performance of this endpoint is adequate so improving the performance is not high priority